### PR TITLE
[pt-r2.1] cherry-pick newer change in nightly repo of #976, #978, #979

### PR DESCRIPTION
**Pull request by ManfeiBai**
# Description

cherry-pick newer change in nightly repo of #976, #978, #979

# Tests

Please describe the tests that you ran on TPUs to verify changes.

**Instruction and/or command lines to reproduce your tests:** 
./scripts/run-oneshot.sh -t pt-nightly-llama2-t-train-spmd-func-v4-8-1vm
./scripts/run-oneshot.sh -t pt-2.1-llama2-t-train-spmd-func-v4-8-1vm
./scripts/run-oneshot.sh -t pt-2.1-lightning-sd-model-pjrt-func-v4-8-1vm

**List links for your tests (use go/shortn-gen for any internal link):**
nightly-llama2-train : http://shortn/_xrTJyNwmEO
2.1-llama2-train : http://shortn/_LZoDmiuVQP
sd-model : http://shortn/_5SWUDuKHav

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run one-shot tests and provided workload links above if applicable. 
- [x] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

**Comment by ManfeiBai** [comment]
Hi, @gkroiz, Stable Diffusion model test failed locally in one-shot test, do you have any solution for these failure? sd-model log: http://shortn/_5SWUDuKHav

File: tests/pytorch/r2.1/llama2-model.libsonnet [hunk 1]
```diff
@@ -160,7 +218,8 @@ local utils = import 'templates/utils.libsonnet';
   },
 
   configs: [
-    llama2_inference + v4_8 + common.Functional + timeouts.Hours(3) + pjrt,
-    llama2_training + v4_8 + common.Functional + timeouts.Hours(3) + hf,
+    llama2_inference + v4_8 + common.Functional + timeouts.Hours(3) + infer,
+    // llama2_training + v4_8 + common.Functional + timeouts.Hours(3) + fsdp,
```
Comment (wonjoolee95) [inline tests/pytorch/r2.1/llama2-model.libsonnet:222]: Do we not want `fsdp`? If not, we can just remove it.

Comment (ManfeiBai) [reply tests/pytorch/r2.1/llama2-model.libsonnet:222]: thanks for pointing it, we don't want fsdp here for now, will delete

Comment (ManfeiBai) [reply tests/pytorch/r2.1/llama2-model.libsonnet:222]: For confirm, do we want to delete config command only and keep setup of fsdp, or we want to delete everything about fsdp

if we keep setup of fsdp here, we could bring it back easily in the future

Comment (wonjoolee95) [reply tests/pytorch/r2.1/llama2-model.libsonnet:222]: Let's just remove it, we can always just refer to this PR if we want to bring it back.

Comment (ManfeiBai) [reply tests/pytorch/r2.1/llama2-model.libsonnet]: Thanks, updated

**Review by wonjoolee95 — APPROVED** [review]

**Pull request merged**